### DISCUSSION TITLE
Booting iPXE from syslinux.efi

### DISCUSSION THREAD

**Discussion by gvarenni** [ipxe/ipxe#657] [Q&A]
Hi.

I'm network booting a UEFI-based system with syslinux.efi. Is it possible to chain iPXE from syslinux.efi i.e. syslinux.efi --> iPXE? If so, which iPXE image should I use? I tried ipxe.lkrn but it loads it and then reboots automatically. If I use the ipxe.efi image, it says the image is not right (I think because syslinux cannot chain an EFI binary).

Thanks,
GV

**Comment by NiKiZe** [comment]
You need to run an .efi build of ipxe when in efi mode. A lkrn relies on pcbios calls (which don't exist in efi mode).
So you need to have syslinux be able to run efi binaries.

The best testcase is generally to have bootloader a run shell.efi to verify the process.

Remember that x86 and x86_64 is different binaries, so you need to match there as well.

**Comment by benthetechguy** [comment]
SYSLINUX cannot chain EFI binaries, and ipxe.lkrn only understands BIOS syscalls, unfortunately.
More info can be found [here](https://github.com/ipxe/wimboot/issues/28) where I tried to do the same thing with wimboot, which is in a similar situation as iPXE.

**Reply by gvarenni** [reply]
Gotcha. I can do the opposite, though i.e. local ipxe.efi first and chain syslinux.efi to it, right?

**Reply by benthetechguy** [reply]
syslinux.efi expects there to be a PXE stack with SNP and all that waiting for it, so after loading from iPXE it just hangs waiting for a connection from which to load its config file.

**Reply by NiKiZe** [reply]
To do anything other then bios calls from lkrn you need full OS. iPXE should provide full working SNP stack.

**Reply by gvarenni** [reply]
syslinux.efi does seem to work on top of iPXE. The thing is that syslinux.efi seems to be fundamentally buggy, whether you load it directly or thru iPXE. Even though using syslinux would have been preferred, as we would maintain the same configuration pxelinux.cfg/default for BIOS and UEFI, there are just too many differences between UEFI and BIOS and iPXE is so much better than pxelinux.

**Reply by benthetechguy** [reply]
> using syslinux would have been preferred, as we would maintain the same configuration pxelinux.cfg/default for BIOS and UEFI

For me, it was easier on iPXE than SYSLINUX to use a single config for both BIOS and UEFI. Now, I'm even using the same config for x86 and ARM. I was stubbornly staying on SYSLINUX for a long time until I ran into the same situation as you; the grass is much greener on this side.

**Reply by amdhugo** [reply]
how to use syslinux to load an arm window os? are you able to do so?

**Comment by stappersg** [comment]
On Wed, Jun 01, 2022 at 09:09:11PM -0700, benthetechguy wrote:
>  the grass is much greener on this side.

Yes.
And it matches the timelime where iPXE was written after syslinux.
(consider iPXE the better syslinux (or pxelinux))
